pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -403,7 +403,6 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 		// instead of disabling the timeout which is Istio's default behavior.
 		if action.Timeout.AsDuration().Nanoseconds() > 0 {
 			action.MaxStreamDuration = &route.RouteAction_MaxStreamDuration{
-				MaxStreamDuration:    action.Timeout,
```

Comment:
For older proxy, this will not work. BTW, what's the side effect if not removed?

Reply:
Why it wont work with older proxy? This was added only in 1.10. Older proxies does not have that bug which was fixed by https://github.com/envoyproxy/envoy/commit/ffa1680ae2619c8339733b9bd7071fc06f5179c9 and timeout works correctly.  The side affect is instead of returning 504 when timeout happens this returns 408. See the linked issue